Print shell version on Travis

lukechilds opened the pull request:
This is really helpful for debugging tests that pass locally but fail on Travis.

ljharb commented:
It seems like `sh` and `dash` don't support `--version`.

lukechilds commented:
Yep, looking into alternative ways to get the version.

I can get the `dash` version from `dpkg` but for `sh` I just get `package 'sh' is not installed and no information is available`.

Does `sh` even have versions?

ljharb commented:
I suspect they might not. I think you can just change this to `$SHELL --version || echo "'$SHELL --version' not supported"` though? (seems to be no need for `eval`)

lukechilds commented:
Nice, for some reason cos `$SHELL` is a var I assumed it'd need to be `eval`'d.

I've updated it to `$SHELL --version 2> /dev/null || dpkg -s $SHELL 2> /dev/null || which $SHELL` which now also gives version data for `dash`. Resorts to just dumping the shell path if both `--version` and `dpkg` fail (`sh`) which may still be helpful.

I actually did this before reading your reply above, if you think this is a bit too verbose and you'd rather I just changed it to the command you mentioned (with `2> /dev/null` on `--version`) I'm happy to do that.

ljharb commented:
No, this is great, thanks - more information is better. Let's make sure tests pass, then it's good to go.

lukechilds commented:
Just out of interest, in Travis before the shell version that I've just added it runs `bash -c 'echo $BASH_VERSION'` for all tests regardless of shell. Is this something Travis does on it's own or are we doing this? I had a look through the source and couldn't find anything on our end.

ljharb commented:
I think it's something travis does, since bash is the default shell on the image.
